Job Titles Matching (ish) 2.0

arahayrabedian opened the pull request:
Same as (closed) #18 to force github not to be an idiot.

Basically, map a job title string to a sorted, stemmed job title string and job modifier string, have not yet done one-hot/binary encoding, will need to borrow @utkuozbulak for that plus my brain is fried:

example of what we do here:
```
In [2]: get_stemmed_sorted_role_and_modifiers("some old granny nurse who takes care of things")
Out[2]: (['nurs'], ['care', 'old', 'take'])

In [2]: get_stemmed_sorted_role_and_modifiers("car salesperson")
Out[2]: ([], ['car'])

In [3]: get_stemmed_sorted_role_and_modifiers("car sales")
Out[3]: (['sale'], ['car'])

In [4]: get_stemmed_sorted_role_and_modifiers("car salesman")
Out[4]: ([], ['car'])
```

basically, we compare stemmed job titles to stemmed official jobs list and see what fits and take that, did some dirty stuff in the mapping that i'll need to go over when encoding to an ML-able format.

it's PR'ing back in to the titles branch so we can finish off and continue work there before merging to master, but i need somebody to have a look first.

also had to make some minor modifications to the oh-so-ancient official job title list, even with a fairly aggressive stemmer.

alexdy2007 committed:
list of towns with population above 50000

alexdy2007 committed:
list of towns with population above 50000

alexdy2007 committed:
Added cities with stop words for title analysis, started title cleaning, Added class to extract data from files to prevent replication of code

alexdy2007 committed:
Added cities with stop words for title analysis, started title cleaning, Added class to extract data from files to prevent replication of code

alexdy2007 committed:
pep8 cleanup

alexdy2007 committed:
pep8 cleanup

charlienewey committed:
Merge branch 'master' into titles

alexdy2007 committed:
Merge branch 'titles' of github.com:soton-data-mining/job-salary-prediction into titles

alexdy2007 committed:
rewrote stop word remover for titles

alexdy2007 committed:
rename job_titles.py to pre_processing_job_titles for clarity

alexdy2007 committed:
removed unecessary file paths for pre_processed_job_title

alexdy2007 committed:
added special chars i.e * '' ~ to stop word list

alexdy2007 committed:
boroughs of london added to cities stop words

arahayrabedian committed:
move random data in to preprocessing_data folder, fix relevant paths

arahayrabedian committed:
add relevant __init__ files

alexdy2007 committed:
data getter for actual jobs added

alexdy2007 committed:
code to split actual job list into job and modifer

arahayrabedian committed:
wow, holy crap, so much done, still needs some work to get in to vector form now

arahayrabedian committed:
Merge branch 'master' into titles_matching

Bring our branch up to date with master

Conflicts:
	requirements.txt

arahayrabedian committed:
fix up some pep8 spacing issues

arahayrabedian committed:
So 1) return in an appropriate format to feed to the learning
 machine and 2) fix the return value to create the sentence string
 instead of as vector of words

utkuozbulak commented:
Hey ! Charles merged my last PR, the binary and hot encoder can now accept lists too, can you also include binary encoded version of the list in main ?

utkuozbulak commented on src/job_titles_feature_extraction.py:
```diff
@@ -0,0 +1,143 @@
+    namely, remove cities and stop words.
+    :return: A string with each word separated by a space for the most part. slashes may live.
+    """
+    job_titles = DataGetter.get_raw_training_data("Title")
```
Can you pass the feature from main to the function instead of reading it all over again ? We already read the data once in main and pass features to functions to clean things up, titles are already in ` train_raw_data`

utkuozbulak commented on src/main.py:
```diff
@@ -31,6 +34,11 @@
 
+    # get stemmed/sorted job title and job title modifiers
+    train_job_titles, train_job_title_modifiers = get_stemmed_sentences(
+        pre_process_job_titles()
```
Related to the other comment: `pre_process_job_titles(title_feature)`  read from `train_raw_data`

I think its better if we follow the same cleaning pattern for all features.
Single data read in main --> pass features to functions --> (Clean the feature) --> return cleaned data to main as list

arahayrabedian replied:
I only moved that stuff around, but will look in to unifying that, it makes sense.

utkuozbulak replied:
Great ! Poke me if you need something :heart:

utkuozbulak commented on src/data_extractor/data_getter.py:
```diff
@@ -0,0 +1,75 @@
+    UNIQUE_JOB_DATA = CURRENT_DIR + "/../../preprocessing_data/job_roles_unique.txt"
+
+    @classmethod
+    def get_towns(cls):
```
Do we have plans to use this later ?

arahayrabedian replied:
@alexdy2007 ^^

arahayrabedian replied:
@alexdy2007 ^^

utkuozbulak commented:
Also, this looks like an extended version of #12 ? Can/Should we delete #12 ?

arahayrabedian commented:
yeah, it builds off the work done in #12 , intention was to merge this back in, do some more work there then merge in to master, but the way it's going i think that yeah, we could probably kill that PR, @alexdy2007 : objections to this?

andreaseliasson commented:
Looks good 👍

alexdy2007 committed:
added few words to stop word list and changed way word tokenisation occurs in title feature extraction

arahayrabedian committed:
Merge branch 'master' into titles_matching

Conflicts:
	requirements.txt
	src/main.py

arahayrabedian committed:
move to project-standard reading of data from csv file using
pandas, internally still uses DataGetter for stop words.

also actually legit binary encodes stuff now.

utkuozbulak commented on src/data_extractor/data_getter.py:
```diff
@@ -0,0 +1,75 @@
+            return words
+
+    @classmethod
+    def get_raw_training_data(cls, header=None):
```
Since you are reading from main now, is this function necessary ?

utkuozbulak commented on src/data_extractor/data_getter.py:
```diff
@@ -0,0 +1,75 @@
+            return jobs
+
+
+if __name__ == "__main__":
```
DataGetter.get_towns() is not being used now(is it ?), is this main necessary here ?

arahayrabedian committed:
- unify data_extractor, DRY and whatnot
 - remove now unused get_raw_training_data method

arahayrabedian commented:
changed base from `titles` to `master`, this PR now contains everything from #12 as well due to that change. sorry for the confusion.

arahayrabedian commented:
@alexdy2007 - please ensure you're happy with the state of `DataGetter`, I made some changes
@utkuozbulak - addressed your comments
@andreaseliasson - would you mind putting that as an approval? - i changed the PR's base branch

alexdy2007 commented:
I'm happy with the changes to DataGetter, looks good to me.

utkuozbulak commented:
Nice 😍

utkuozbulak reviewed (approved):
I think we will play with the job titles but this is a good start.

A notice for everyone, will push a cleanup on main and separate encodings etc to another py file as @blanche 's request

arahayrabedian commented:
@utkuozbulak yeah, I think job titles has a LOT of room to play with. This isn't even it's final form.
